lib/features/campaigns/screens/teams_screen.dart

```diff
@@ -1,17 +1,148 @@
+        ),
+        child: Stack(
+          children: [
+            Align(
+              alignment: Alignment.centerLeft,
+              child: Text(t.campaigns.team.create_team, style: theme.textTheme.bodyLarge),
+            ),
+            Align(
+              alignment: Alignment.centerRight,
+              child: Icon(Icons.chevron_right, color: theme.textTheme.bodyLarge?.color),
+            ),
+          ],
+        ),
+      ),
+    );
+    rows.add(
+      FutureBuilder(
+        future: AuthRepository().getUserInfo(),
+        builder: (context, snapshot) {
+          if (!snapshot.hasData) return SizedBox.shrink();
+
+          var userInfo = snapshot.data!;
+          if (userInfo.isCampaignManager()) {
+            return rowCreateTeam;
+          }
+          return SizedBox.shrink();
+        },
       ),
     );
+
+    return Column(children: rows);
+  }
+
+  void _beginCreateNewTeam(BuildContext context) async {
+    final theme = Theme.of(context);
+
+    var newTeamWidget = NewTeamBasicInfoWidget();
+    var newDetails = await showModalBottomSheet<NewTeamDetails>(
+      context: context,
+      builder: (context) => newTeamWidget,
+      isScrollControlled: false,
+      isDismissible: true,
+      backgroundColor: theme.colorScheme.surface,
+    );
+
+    if (context.mounted) {
+      if (newDetails != null) {
+        _showStepDivisionSelect(newDetails, context);
+      }
+    }
+  }
+
+  void _showStepDivisionSelect(NewTeamDetails newTeamDetails, BuildContext context) async {
+    final theme = Theme.of(context);
+
+    var newTeamWidget = NewTeamSelectDivisionWidget(newTeamDetails: newTeamDetails);
+    var newDetails = await showModalBottomSheet<NewTeamDetails>(
+      context: context,
+      builder: (context) => newTeamWidget,
+      isScrollControlled: false,
+      isDismissible: true,
+      backgroundColor: theme.colorScheme.surface,
+    );
+
+    if (context.mounted) {
+      if (newDetails != null) {
+        _showStepTeamLeadSelect(newDetails, context);
+      }
+    }
+  }
+
+  void _showStepTeamLeadSelect(NewTeamDetails newTeamDetails, BuildContext context) async {
+    final theme = Theme.of(context);
+    NewTeamDetails? newDetails = newTeamDetails;
+    if (!newDetails.selfJoin) {
+      var newTeamWidget = NewTeamSelectTeamLeadWidget(newTeamDetails: newTeamDetails);
+      newDetails = await showModalBottomSheet<NewTeamDetails>(
+        context: context,
+        builder: (context) => newTeamWidget,
+        isScrollControlled: false,
+        isDismissible: true,
+        backgroundColor: theme.colorScheme.surface,
+      );
+    }
+
+    if (context.mounted) {
+      if (newDetails != null) {
+        _showStepTeamMemberSelect(newDetails, context);
+      }
+    }
+  }
+
+  Future<void> _showStepTeamMemberSelect(NewTeamDetails newTeamDetails, BuildContext context) async {
+    final theme = Theme.of(context);
+    while (true) {
```

Comment:
Instinctively I don't super duper like a while true combined with a break statement but I see how it works here. As an alternative, could we also have a variable that is true and _set it to false_ instead of the `break`?